packages/jellyfish-api-core/__tests__/category/icxorderbook/listOrders.test.ts

```diff
@@ -0,0 +1,328 @@
+      chainTo: 'BTC',
+      ownerAddress: accountDFI,
+      receivePubkey: '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941',
+      amountFrom: new BigNumber(20),
+      orderPrice: new BigNumber(0.01)
+    }
+    let result: ICXGenericResult = await client.icxorderbook.createOrder(order, [])
+    const createOrderTxId = result.txid
+    await container.generate(1)
+
+    // create second order - maker
+    const order2: ICXOrder = {
+      tokenFrom: idDFI,
+      chainTo: 'BTC',
+      ownerAddress: accountDFI,
+      receivePubkey: '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941',
+      amountFrom: new BigNumber(15),
+      orderPrice: new BigNumber(0.01)
+    }
+    result = await client.icxorderbook.createOrder(order2, [])
+    const createOrder2TxId = result.txid
+    await container.generate(1)
+
+    // list ICX orders anc check
+    let orders: Record<string, ICXOrderInfo| ICXOfferInfo> = await client.icxorderbook.listOrders()
+    await checkDFISellOrderDetails(container, order, createOrderTxId, orders as Record<string, ICXOrderInfo>)
+    await checkDFISellOrderDetails(container, order2, createOrder2TxId, orders as Record<string, ICXOrderInfo>)
+
+    // now close createOrder2TxId
+    await client.icxorderbook.closeOrder(createOrder2TxId)
+    await container.generate(1)
+
+    // list ICX orders and check
+    orders = await client.icxorderbook.listOrders()
+    await checkDFISellOrderDetails(container, order, createOrderTxId, orders as Record<string, ICXOrderInfo>)
+
+    // list ICX orders with ICXListOrderOptions.closed=true and check
+    orders = await client.icxorderbook.listOrders({ closed: true })
+    const orderInfoRetrived = orders as Record<string, ICXOrderInfo>
+    expect(orderInfoRetrived[createOrder2TxId].status).toStrictEqual(ICXOrderStatus.CLOSED)
+    expect(orderInfoRetrived[createOrder2TxId].ownerAddress).toStrictEqual(order2.ownerAddress)
+
+    // make offer to partial amout 10 DFI - taker
+    const offer: ICXOffer = {
+      orderTx: createOrderTxId,
+      amount: new BigNumber(0.10), // 0.10 BTC = 10 DFI
+      ownerAddress: accountBTC
+    }
+    // const accountBTCBeforeOffer = await container.call('getaccount', [accountBTC, {}, true])
+    result = await client.icxorderbook.makeOffer(offer, [])
+    const makeOfferTxId = result.txid
+    await container.generate(1)
+
+    // make second offer to partial amout 10 DFI - taker
+    const offer2: ICXOffer = {
+      orderTx: createOrderTxId,
+      amount: new BigNumber(0.10), // 0.10 BTC = 10 DFI
+      ownerAddress: accountBTC
+    }
+    result = await client.icxorderbook.makeOffer(offer2, [])
+    const makeOffer2TxId = result.txid
+    await container.generate(1)
+
+    // const accountBTCAfterOffer = await container.call('getaccount', [accountBTC, {}, true])
+
+    // NOTE(surangap): why below check is failing?
+    // check fee of 0.02 DFI has been reduced from the accountBTCBeforeOffer[idDFI]
```

Comment:
if both offers are successful, then the taker fee of 0.02 DFI should be reduced from the balance. instead, only 0.01 was reduced.